src/features/hotspots/checklist/HotspotChecklist.tsx

```diff
@@ -27,18 +27,23 @@ const HotspotChecklist = ({ hotspot, witnesses }: Props) => {
   const spacing = useSpacing()
   const { t } = useTranslation()
   const {
-    heliumData: { blockHeight },
-    hotspotChecklist: {
-      challengerTxn,
-      challengeeTxn,
-      witnessTxn,
-      dataTransferTxn,
-      loadingActivity,
-    },
-  } = useSelector((state: RootState) => state)
+    challengerTxn,
+    challengeeTxn,
+    witnessTxn,
+    dataTransferTxn,
+    loadingActivity,
+  } = useSelector((state: RootState) => state.hotspotChecklist)
+  const blockHeight = useSelector(
+    (state: RootState) => state.heliumData.blockHeight,
+  )
+  const checklistEnabled = useSelector(
+    (state: RootState) => state.features.checklistEnabled,
+  )
 
   useEffect(() => {
-    dispatch(fetchChecklistActivity(hotspot.address))
+    if (checklistEnabled) {
+      dispatch(fetchChecklistActivity(hotspot.address))
+    }
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [])
```

Comment:
does `checklistEnabled` need to be included in the hook's dependencies?

Reply:
Yeah good point, since before that was only called at start but if the flag gets enabled we should call it. Ill add all the dependencies now